test/extended/storage/inline.go

```diff
@@ -0,0 +1,391 @@
+package storage
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/google/uuid"
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	k8simage "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	configv1 "github.com/openshift/api/config/v1"
+	securityv1 "github.com/openshift/api/security/v1"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+const (
+	csiInlineVolProfileLabel = "security.openshift.io/csi-ephemeral-volume-profile"
+	csiSharedResourceDriver  = "csi.sharedresource.openshift.io"
+	podSecurityEnforceError  = "has a pod security enforce level that is lower than"
+)
+
+// This is [Serial] because it modifies a CSIDriver object that is used by multiple tests.
+var _ = g.Describe("[sig-storage][Feature:CSIInlineVolumeAdmission][Serial]", func() {
+	defer g.GinkgoRecover()
+	var (
+		ctx                  = context.Background()
+		baseDir              = exutil.FixturePath("testdata", "storage", "inline")
+		secret               = filepath.Join(baseDir, "secret.yaml")
+		csiSharedSecret      = filepath.Join(baseDir, "csi-sharedsecret.yaml")
+		csiSharedRole        = filepath.Join(baseDir, "csi-sharedresourcerole.yaml")
+		csiSharedRoleBinding = filepath.Join(baseDir, "csi-sharedresourcerolebinding.yaml")
+		sccVolumeToggle      *SCCVolumeToggle
+
+		beforeEach = func(oc *exutil.CLI) {
+			if !isTechPreviewNoUpgrade(oc) {
+				g.Skip("this test is only expected to work with TechPreviewNoUpgrade clusters")
+			} else {
+				// TODO: remove the SCCVolumeToggle when CSI volumes are allowed by the default SCC's (i.e. after TechPreview)
+				g.By("adding restricted-v2 SCC permission to use inline CSI volumes")
+				sccVolumeToggle = NewSCCVolumeToggle(oc, "restricted-v2", securityv1.FSTypeCSI)
+				sccVolumeToggle.Enable()
+			}
+			exutil.PreTestDump()
+
+			// create the secret to share in a new namespace
```

Comment:
in future testing PRs, please avoid using `oc` commands to create items. This actually forks and `oc` process which is more difficult to debug after the fact when it fails.  Prefer to use library-go resourceread methods combined with the `oc.kubernetesclient.core.v1.secret.create` style.